packages/components/src/form-token-field/suggestions-list.tsx

```diff
@@ -52,16 +51,10 @@ export function SuggestionsList< T extends string | { value: string } >( {
 						onlyScrollIfNeeded: true,
 					}
 				);
-				id = window.setTimeout( () => {
+				requestAnimationFrame( () => {
 					setScrollingIntoView( false );
-				}, 100 );
+				} );
 			}
-
-			return () => {
-				if ( id !== undefined ) {
-					window.clearTimeout( id );
-				}
```

Comment:
I'm not sure there's a point in running `cancelAnimationFrame` in the effect's return function, and so for now I've avoided it.

Reply:
I didn't get a chance to review properly today, unfortunately, but hopefully should get a bit more time next week once some things for 6.1 Beta 3 are ready.

From a quick skim of the failing unit test, though, it's reporting:

```
["Warning: Can't perform a React state update on an unmounted component. This is a no-op, but it indicates a memory leak in your application. To fix, cancel all subscriptions and asynchronous tasks in %s.%s", "a useEffect cleanup function", "
```

I wonder if that warning is showing up because the `requestAnimationFrame` is still firing? If so, I wonder if we might need to grab the id returned from `requestAnimationFrame` and call `cancelAnimationFrame` so that it doesn't attempt to update state after unmounting?

Reply:
Makes sense, done in https://github.com/WordPress/gutenberg/pull/44573/commits/566b233e76baaf66eedceaae75f6223560bb67a6